### PR TITLE
refactor: streamline create & edit account pages

### PR DESCRIPTION
**Pull request by logan-r**
### Tickets:

- HCK- #841
- HCK- #847

### List of changes:

- Refactor account forms to use hooks
- Remove ManageAccountContainer 
- Break out form specific code from ManageAccountContainer into ManageAccountForm
- Move ManageAccountContainer's page specific styling (e.g. background images) and <Head>-level elements to either CreateAccountPage or EditAccountPage
- Implemented seperate backgrounds on Create and Edit account pages, without need for many if statements on ManageAccountModes
- Make background images more responsive on both pages, fixing mobile background/form elements interference glitch

### Type of change:

Please delete options that aren't relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### PR Checklist:

- [x] Merged `develop` branch (before testing)
- [x] Linted my code locally
- [ ] Listed change(s) in the Changelog
- [x] Tested all links in project relevant browsers
- [x] Tested all links on different screen sizes
- [x] Referenced all useful info (issues, tasks, etc)

### Screenshots:

[Image from Gyazo](https://gyazo.com/6b650e980a3b46d24860a4fe19ea40a2)

[Image from Gyazo](https://gyazo.com/e16ffdd2b58af46b2374c67f1f5f32c0)

### CODE REVIEW

**Comment by pierreTklein** [comment]
Can you confirm that you were able to successfully:

1. Edit an account,
2. Create an account,
3. Trigger invalid form check?

**Comment by logan-r** [comment]
Fixed a bug with the birthdate field not being filled in when in edit mode.
Can now complete those 3 tasks.

File: src/features/Account/ManageAccountForm.tsx [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+    password: getNestedAttr(props, ['location', 'state', 'password']) || '',
+    phoneNumber: '',
+    pronoun: '',
+    gender: '',
+    dietaryRestrictions: [],
+  })
+
+  // If this is an edit form, load current account's data as default value
+  // for account details
+  useEffect(() => {
```
Comment (pierreTklein) [inline src/features/Account/ManageAccountForm.tsx:75]: I've never used react hooks before, so I'm not too sure what the potential downsides of using an async function is. I found this on stack-overflow: https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret

Can you confirm whether this is an issue for this implementation?

File: src/features/Account/ManageAccountForm.tsx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    (async () => {
+      if (props.mode === ManageAccountModes.EDIT) {
+        try {
+          const response = await Account.getSelf();
+          const newAccountDetails = response.data.data;
+          newAccountDetails.birthDate = date2input(newAccountDetails.birthDate);
```
Comment (pierreTklein) [inline src/features/Account/ManageAccountForm.tsx:81]: What's the type of `newAccountDetails.birthDate`? Is it either an input or a date?

File: src/features/Account/ManageAccountForm.tsx [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+    // or update user's existing account to match form data
+    if (props.mode === ManageAccountModes.CREATE) {
+      handleCreate(formattedDetails);
+    } else if (props.mode === ManageAccountModes.EDIT) {
+      handleEdit(formattedDetails, values.password, values.newPassword);
+    }
+
+    // Once submitted, redirect user to appropriate page
+    if (isSponsor(accountDetails)) {
+      history.push(FrontendRoute.CREATE_SPONSOR_PAGE);
```
Comment (pierreTklein) [inline src/features/Account/ManageAccountForm.tsx:131]: Can you test this behavior with a sponsor account that is editing their profile? What does this do?

Comment (logan-r) [reply src/features/Account/ManageAccountForm.tsx:131]: I missed this case when redoing the redirect code, this should be `if (props.mode === ManageAccountModes.CREATE && isSponsor(accountDetails))`

File: src/features/Account/ManageAccountForm.tsx (hunk 1 above)
Comment (logan-r) [reply src/features/Account/ManageAccountForm.tsx:75]: Basically useEffect should either return nothing or a non-async function that cleans up (e.g. unsubscribes from updates). We return nothing here, so no issues.

Comment (pierreTklein) [reply src/features/Account/ManageAccountForm.tsx:75]: Great 👍

File: src/features/Account/ManageAccountForm.tsx (hunk 3 above)
Comment (pierreTklein) [reply src/features/Account/ManageAccountForm.tsx:131]: Feel free to resolve after you implement the mentioned change & then test.

File: src/features/Account/ManageAccountForm.tsx (hunk 1 above)
Comment (logan-r) [reply src/features/Account/ManageAccountForm.tsx:75]: * The function passed to useEffect should either return

File: src/features/Account/ManageAccountForm.tsx (hunk 2 above)
Comment (logan-r) [reply src/features/Account/ManageAccountForm.tsx:81]: It's a string before and after the date2input call

Comment (pierreTklein) [reply src/features/Account/ManageAccountForm.tsx:81]: Cool. What's the purpose of `date2input` again?

Comment (logan-r) [reply src/features/Account/ManageAccountForm.tsx:81]: Turns `"1998-11-30T04:00:00.000Z"` into `"11301998"`

Comment (pierreTklein) [reply src/features/Account/ManageAccountForm.tsx:81]: Cool. Resolving.

**Pull request merged**